here-naksha-lib-hub/src/main/java/com/here/naksha/lib/hub/NakshaHub.java

```diff
@@ -267,11 +271,18 @@ public NakshaHub(
         throw unchecked(new Exception(
             "Exception fetching storage details for id " + storageId + ". " + er.message, er.exception));
       }
-      final Storage storage = readFeatureFromResult(result, Storage.class);
-      if (storage == null) {
-        throw unchecked(new Exception("No storage found with id " + storageId));
-      }
-      return storage.newInstance(this);
+      final Storage storage =
+          Objects.requireNonNull(readFeatureFromResult(result, Storage.class), "Storage not found in result");
```

Comment:
eventhough minor thing, but let's retain the previous message (which prints the storageId)
("No storage found with id " + storageId)

Reply:
done

Reply:
done